### ISSUE TITLE
NgModel within NgFor results in exception 'Cannot reassign a variable binding'

### ISSUE THREAD

**Issue by ralfe-software** [angular/angular#8086]
I have a list which a user should fill/update from predefined items. For this i use NgFor and create HTML-select which binds to the current loop iteration value via NgModel and define the possible option-values.

**Steps to reproduce and a minimal demo of the problem**
Plunker:
https://plnkr.co/edit/RBzuCaysmXFR0EVZqQO2?p=info

Let it run and exception 'Cannot reassign a variable binding' occurs. Remove from select-tag the ngModel two-way-binding (app.ts, line 13) the sample runs but the list (named buyList) does not change.

**Current behavior**
Exception 'Cannot reassign a variable binding' occurs.

**Expected/desired behavior**
No exception should happen and bound list entries gets updated by user selection.

**Other information**
#4843 made it possible to bind to objects in switch/option tags and created my expectation that it should work.

**Comment by GregOnNet** [comment]
Hi Ralf,

in my opinion it is not common to try to reassign an item of a list within the `ngFor`-directive.
I build a [Plunkr demo](https://plnkr.co/edit/Q5flNFG9dpDo6xb6gbVL?p=preview) for you to get around this problem.
Please notice that I hacked this demo as workaround. :smirk: 

You may want to introduce a own component structure handling updates of your data model.

Kinds
Greg

**Comment by kara** [comment]
@ralfe-software It's not possible to create a two-way binding to a local template variable. You'll want to bind to your domain model directly. e.g.

```html
<!-- do this -->
<div *ngFor="let item of items; let i=index">
   <input [(ngModel)]="items[i]">
</div>

<!-- instead of this -->
<div *ngFor="let item of items">
   <input [(ngModel)]="item">
</div>
```

You should still be able to bind objects to options using ngValue, though.

**Comment by ralfe-software** [comment]
Thanks for the clarification.

Greetings, Ralf


Von: Kara <notifications@github.com><mailto:notifications@github.com>
Antworten: angular/angular <reply@reply.github.com><mailto:reply@reply.github.com>
Datum: 12. Oktober 2016 at 23:13:45
An: angular/angular <angular@noreply.github.com><mailto:angular@noreply.github.com>
CC: Ralf Ehlert <ralf@ralfe-software.net><mailto:ralf@ralfe-software.net>, Mention <mention@noreply.github.com><mailto:mention@noreply.github.com>
Betreff:  Re: [angular/angular] NgModel within NgFor results in exception 'Cannot reassign a variable binding' (#8086)


Closed #8086<https://github.com/angular/angular/issues/8086>.
